# HG changeset patch
# User Rik <rik@octave.org>
# Date 1347732607 25200
#      Sat Sep 15 11:10:07 2012 -0700
# Node ID c7fd43f5a89dc4dc916934fe6e63de0c75df3e7d
# Parent  1ddf5772fb1a94b446ae177b375a8dab8908bfbe
build: Re-arrange libgui Makefile.am to follow same ordering as liboctave.

* libgui/Makefile.am: Re-arrange order of definitions and rules to follow
same ordering as liboctave.

* libgui/qterminal-module.mk: Re-arrange definition to keep line length
less than 80 characters.

* libgui/src/module.mk: Re-arrange order of definitions and rules to follow
same ordering as in liboctave module.mk files.

diff --git a/libgui/Makefile.am b/libgui/Makefile.am
--- a/libgui/Makefile.am
+++ b/libgui/Makefile.am
@@ -31,34 +31,34 @@ QT_LIBS = -lQtCore -lQtGui -lQtNetwork -
 QT_LDFLAGS = -L$(QT_LIBDIR)
 
 MOC_CPPFLAGS =
 
 octlib_LTLIBRARIES = liboctgui.la
 
 EXTRA_DIST = default-qt-settings
 
-BUILT_SOURCES =
+CLEANFILES =
 
-CLEANFILES =
+BUILT_SOURCES =
 
 noinst_HEADERS =
 
 noinst_LTLIBRARIES =
 
 include src/module.mk
 include qterminal-module.mk
 
-## libocgui merely collects of compiled convenience libraries.
+## liboctgui merely collects a bunch of compiled convenience libraries.
 ## It has no source code itself.
 liboctgui_la_SOURCES = 
 
 include link-deps.mk
 
-# Dummy C++ source to cause C++ linking.
+# Dummy C++ source to force C++ linking.
 nodist_EXTRA_liboctgui_la_SOURCES = dummy.cc
 
 liboctgui_la_LIBADD = \
   qterminal/libqterminal.la \
   src/libgui-src.la \
   $(top_builddir)/libinterp/liboctinterp.la \
   $(top_builddir)/liboctave/liboctave.la \
   $(LIBOCTGUI_LINK_DEPS)
diff --git a/libgui/qterminal-module.mk b/libgui/qterminal-module.mk
--- a/libgui/qterminal-module.mk
+++ b/libgui/qterminal-module.mk
@@ -45,17 +45,19 @@ qterminal_libqterminal_la_CFLAGS = $(AM_
 qterminal_libqterminal_la_CXXFLAGS = $(AM_CXXFLAGS)
 
 if WIN32_TERMINAL
 
 qterminal_libqterminal_la_SOURCES = \
   qterminal/libqterminal/win32/QTerminalColors.cpp \
   qterminal/libqterminal/win32/QWinTerminalImpl.cpp
 
-qterminal_libqterminal_la_MOC += qterminal/libqterminal/win32/moc-QWinTerminalImpl.cc
+qterminal_libqterminal_la_MOC += \
+  qterminal/libqterminal/win32/moc-QWinTerminalImpl.cc
+
 qterminal_libqterminal_la_CPPFLAGS += -DUNICODE
 
 # This flag is required to let MOC know about Q_OS_WIN32.
 MOC_CPPFLAGS += -DWIN32
 
 else
 
 qterminal_libqterminal_la_SOURCES = \
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -1,102 +1,14 @@
 EXTRA_DIST += \
   src/module.mk \
   src/resource.qrc \
   $(octave_gui_UI) \
   $(octave_gui_ICONS)
 
-octave_gui_MOC = \
-  src/m-editor/moc-file-editor-interface.cc \
-  src/m-editor/moc-file-editor-tab.cc \
-  src/m-editor/moc-file-editor.cc \
-  src/m-editor/moc-find-dialog.cc \
-  src/m-editor/moc-lexer-octave-gui.cc \
-  src/moc-documentation-dockwidget.cc \
-  src/moc-files-dockwidget.cc \
-  src/moc-history-dockwidget.cc \
-  src/moc-main-window.cc \
-  src/moc-octave-qt-event-listener.cc \
-  src/moc-settings-dialog.cc \
-  src/moc-terminal-dockwidget.cc \
-  src/moc-welcome-wizard.cc \
-  src/moc-workspace-model.cc \
-  src/moc-workspace-view.cc \
-  src/octave-adapter/moc-octave-main-thread.cc \
-  src/qtinfo/moc-parser.cc \
-  src/qtinfo/moc-webinfo.cc
-
-octave_gui_UI = \
-  src/settings-dialog.ui \
-  src/welcome-wizard.ui
-
-octave_gui_UI_H = $(patsubst src/%.ui, src/ui-%.h, $(octave_gui_UI))
-
-octave_gui_RC = src/qrc-resource.cc
-
-src_libgui_src_la_SOURCES = \
-  src/documentation-dockwidget.cc \
-  src/files-dockwidget.cc \
-  src/history-dockwidget.cc \
-  src/m-editor/file-editor-tab.cc \
-  src/m-editor/file-editor.cc \
-  src/m-editor/find-dialog.cc \
-  src/m-editor/lexer-octave-gui.cc \
-  src/main-window.cc \
-  src/octave-adapter/octave-event.cc \
-  src/octave-adapter/octave-link.cc \
-  src/octave-adapter/octave-main-thread.cc \
-  src/octave-gui.cc \
-  src/octave-qt-event-listener.cc \
-  src/qtinfo/parser.cc \
-  src/qtinfo/webinfo.cc \
-  src/resource-manager.cc \
-  src/settings-dialog.cc \
-  src/symbol-information.cc \
-  src/terminal-dockwidget.cc \
-  src/welcome-wizard.cc \
-  src/workspace-model.cc \
-  src/workspace-view.cc
-
-nodist_src_libgui_src_la_SOURCES = $(octave_gui_MOC) $(octave_gui_RC)
-
-noinst_HEADERS += \
-  src/documentation-dockwidget.h \
-  src/files-dockwidget.h \
-  src/history-dockwidget.h \
-  src/m-editor/file-editor-interface.h \
-  src/m-editor/file-editor-tab.h \
-  src/m-editor/file-editor.h \
-  src/m-editor/find-dialog.h \
-  src/m-editor/lexer-octave-gui.h \
-  src/main-window.h \
-  src/octave-adapter/octave-event-listener.h \
-  src/octave-adapter/octave-event-observer.h \
-  src/octave-adapter/octave-event.h \
-  src/octave-adapter/octave-link.h \
-  src/octave-adapter/octave-main-thread.h \
-  src/octave-gui.h \
-  src/octave-qt-event-listener.h \
-  src/qtinfo/parser.h \
-  src/qtinfo/webinfo.h \
-  src/resource-manager.h \
-  src/settings-dialog.h \
-  src/symbol-information.h \
-  src/terminal-dockwidget.h \
-  src/welcome-wizard.h \
-  src/workspace-model.h \
-  src/workspace-view.h
-
-CLEANFILES += \
-  $(octave_gui_MOC) \
-  $(octave_gui_UI_H) \
-  $(octave_gui_RC)
-
-BUILT_SOURCES += $(octave_gui_UI_H)
-
 octave_gui_ICONS = \
   src/icons/arrow_right.png \
   src/icons/artsbuilderexecute.png \
   src/icons/bookmark.png \
   src/icons/bp_next.png \
   src/icons/bp_prev.png \
   src/icons/bp_rm_all.png \
   src/icons/bp_toggle.png \
@@ -126,16 +38,99 @@ octave_gui_ICONS = \
   src/icons/star.png \
   src/icons/stop.png \
   src/icons/terminal.png \
   src/icons/undo.png \
   src/icons/up.png \
   src/icons/zoom-in.png \
   src/icons/zoom-out.png
 
+octave_gui_MOC = \
+  src/m-editor/moc-file-editor-interface.cc \
+  src/m-editor/moc-file-editor-tab.cc \
+  src/m-editor/moc-file-editor.cc \
+  src/m-editor/moc-find-dialog.cc \
+  src/m-editor/moc-lexer-octave-gui.cc \
+  src/moc-documentation-dockwidget.cc \
+  src/moc-files-dockwidget.cc \
+  src/moc-history-dockwidget.cc \
+  src/moc-main-window.cc \
+  src/moc-octave-qt-event-listener.cc \
+  src/moc-settings-dialog.cc \
+  src/moc-terminal-dockwidget.cc \
+  src/moc-welcome-wizard.cc \
+  src/moc-workspace-model.cc \
+  src/moc-workspace-view.cc \
+  src/octave-adapter/moc-octave-main-thread.cc \
+  src/qtinfo/moc-parser.cc \
+  src/qtinfo/moc-webinfo.cc
+
+octave_gui_RC = src/qrc-resource.cc
+
+octave_gui_UI = \
+  src/settings-dialog.ui \
+  src/welcome-wizard.ui
+
+octave_gui_UI_H = $(patsubst src/%.ui, src/ui-%.h, $(octave_gui_UI))
+
+BUILT_SOURCES += $(octave_gui_UI_H)
+
+noinst_HEADERS += \
+  src/documentation-dockwidget.h \
+  src/files-dockwidget.h \
+  src/history-dockwidget.h \
+  src/m-editor/file-editor-interface.h \
+  src/m-editor/file-editor-tab.h \
+  src/m-editor/file-editor.h \
+  src/m-editor/find-dialog.h \
+  src/m-editor/lexer-octave-gui.h \
+  src/main-window.h \
+  src/octave-adapter/octave-event-listener.h \
+  src/octave-adapter/octave-event-observer.h \
+  src/octave-adapter/octave-event.h \
+  src/octave-adapter/octave-link.h \
+  src/octave-adapter/octave-main-thread.h \
+  src/octave-gui.h \
+  src/octave-qt-event-listener.h \
+  src/qtinfo/parser.h \
+  src/qtinfo/webinfo.h \
+  src/resource-manager.h \
+  src/settings-dialog.h \
+  src/symbol-information.h \
+  src/terminal-dockwidget.h \
+  src/welcome-wizard.h \
+  src/workspace-model.h \
+  src/workspace-view.h
+
+src_libgui_src_la_SOURCES = \
+  src/documentation-dockwidget.cc \
+  src/files-dockwidget.cc \
+  src/history-dockwidget.cc \
+  src/m-editor/file-editor-tab.cc \
+  src/m-editor/file-editor.cc \
+  src/m-editor/find-dialog.cc \
+  src/m-editor/lexer-octave-gui.cc \
+  src/main-window.cc \
+  src/octave-adapter/octave-event.cc \
+  src/octave-adapter/octave-link.cc \
+  src/octave-adapter/octave-main-thread.cc \
+  src/octave-gui.cc \
+  src/octave-qt-event-listener.cc \
+  src/qtinfo/parser.cc \
+  src/qtinfo/webinfo.cc \
+  src/resource-manager.cc \
+  src/settings-dialog.cc \
+  src/symbol-information.cc \
+  src/terminal-dockwidget.cc \
+  src/welcome-wizard.cc \
+  src/workspace-model.cc \
+  src/workspace-view.cc
+
+nodist_src_libgui_src_la_SOURCES = $(octave_gui_MOC) $(octave_gui_RC)
+
 src_libgui_src_la_CPPFLAGS = \
   $(AM_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   -I$(QT_INCDIR) \
   -I$(QT_INCDIR)/QtCore \
   -I$(QT_INCDIR)/QtGui \
   -I$(QT_INCDIR)/QtNetwork \
   -I$(srcdir)/qterminal/libqterminal \
@@ -155,8 +150,14 @@ src_libgui_src_la_CPPFLAGS = \
   -I$(top_builddir)/libinterp/interpfcn -I$(top_srcdir)/libinterp/interpfcn \
   -I$(top_srcdir)/libinterp/octave-value
 
 src_libgui_src_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 src_libgui_src_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 noinst_LTLIBRARIES += src/libgui-src.la
+
+CLEANFILES += \
+  $(octave_gui_MOC) \
+  $(octave_gui_UI_H) \
+  $(octave_gui_RC)
+
